### PR TITLE
RTL8139 Alloc regs resource

### PR DESCRIPTION
**Pull request by komik0**
With this commit the following changes are introduced:
- resource for rtl8139 regs is added
- software reset is performed
- dirty workaround for `rman_alloc_resource`

Important note:
to proceed with further improvement of this driver, there is needed to close #822 and #820

### CODE REVIEW

File: sys/drv/rtl8139.c [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include <sys/libkern.h>
 #include <sys/devfs.h>
 #include <sys/devclass.h>
+#include "rtl8139_reg.h"
```
Comment (pj1031999) [inline sys/drv/rtl8139.c:10]: Please use headers only from `include` directory.

File: sys/drv/rtl8139.c [hunk 2]
```diff
@@ -21,7 +22,33 @@ static int rtl8139_probe(device_t *dev) {
...
+
+  if (i == RL_TIMEOUT)
+    klog("RTL8139: Failed to reset device!");
+}
+
 static int rtl8139_attach(device_t *dev) {
+  assert(dev->parent->bus == DEV_BUS_PCI);
+  rtl8139_state_t *state = dev->state;
+
+  klog("### RTL8139 DUMMY DRIVER");
```
Comment (pj1031999) [inline sys/drv/rtl8139.c:43]: `klog` in `bus_generic_probe` already do that.

File: sys/drv/rtl8139.c [hunk 3]
```diff
@@ -21,7 +22,33 @@ static int rtl8139_probe(device_t *dev) {
...
+
+  state->regs = bus_alloc_resource_any(dev, RT_MEMORY, 1, RF_ACTIVE);
+
+  if (state->regs == NULL)
+    panic("rtl8139 alloc resource error");
```
Comment (pj1031999) [inline sys/drv/rtl8139.c:48]: In case of failure attach should return with error value.

File: sys/drv/rtl8139_reg.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * Command register.
+ */
+#define RL_CMD_RESET            0x0010
+
+/*
+ * Config 1 register
+ */
+#define RL_CFG1_PWRON           0x00
+
+#define RL_TIMEOUT              (10*1000)
```
Comment (pj1031999) [inline sys/drv/rtl8139_reg.h:19]: Please add include guard.

**Review by pj1031999 — CHANGES_REQUESTED** [review]

File: sys/drv/rtl8139_reg.h (hunk 4 above)
Comment (komik0) [reply sys/drv/rtl8139_reg.h]: ok - done. fyi, only i8259.h and isareg.h take care about guard

File: sys/drv/rtl8139.c (hunk 3 above)
Comment (komik0) [reply sys/drv/rtl8139.c]: thanks, done

File: sys/drv/rtl8139.c (hunk 2 above)
Comment (komik0) [reply sys/drv/rtl8139.c]: thanks, done

File: sys/drv/rtl8139.c (hunk 1 above)
Comment (komik0) [reply sys/drv/rtl8139.c]: done

**Comment by komik0** [comment]
I will upload easier to review version in another PR

**Pull request closed without merging**